src/ngraph_builder.cc

```diff
@@ -2019,6 +2019,35 @@ static Status TranslateRsqrtOp(
       });
 }
 
+static Status TranslateShapeOp(
+    const Node* op, const std::vector<const Tensor*>& static_input_map,
+    Builder::OpMap& ng_op_map) {
+  shared_ptr<ng::Node> ng_input;
+  TF_RETURN_IF_ERROR(GetInputNodes(ng_op_map, op, &ng_input));
+
+  // the shape of the input tensor which will be the value to the Constant Op
+  auto input_shape = ng_input->get_shape();
+
+  // the rank of the input tensor which will be the shape to the Constant Op
+  auto rank = input_shape.size();
+
+  // the inputs to the Constant Op
+  // type
+  ng::element::Type type;
+  TF_RETURN_IF_ERROR(TFDataTypeToNGraphElementType(DataType::DT_INT32, &type));
+
+  // shape
+  auto shape = ng::Shape(1, rank);
+  // values
+  std::vector<int> values(rank);
+  for (int i = 0; i < rank; i++) {
+    values[i] = input_shape[i];
+  }
+  SaveNgOp(ng_op_map, op->name(),
+           make_shared<ng::op::Constant>(type, shape, values));
+  return Status::OK();
```

Comment:
instead of type, lets use ```ng::element::i32``` directly. Then we do not need the lines above:
```
ng::element::Type type;
TF_RETURN_IF_ERROR(TFDataTypeToNGraphElementType(DataType::DT_INT32, &type));
```